src/ui/pages/orders/delivery.page.ts

```diff
@@ -0,0 +1,137 @@
+    delivery.condition && (await this.typeInput.selectOption(delivery.condition));
+    delivery.finalDate && (await this.fillDateInput(delivery.finalDate));
+    if (delivery.condition === DELIVERY_CONDITIONS.PICK_UP) {
+      delivery.address?.country && (await this.countryInput.selectOption(delivery.address.country));
+      return;
+    }
+    delivery.location && (await this.locationInput.selectOption(delivery.location));
+    if (delivery.location === 'Other') {
+      delivery.address?.country && (await this.countryInput.selectOption(delivery.address.country));
+      delivery.address?.city && (await this.cityInput.fill(delivery.address.city));
+      delivery.address?.street && (await this.streetInput.fill(delivery.address.street));
+      delivery.address?.house && (await this.houseInput.fill(delivery.address.house.toString()));
+      delivery.address?.flat && (await this.flatInput.fill(delivery.address.flat.toString()));
+    }
+  }
+
+  @logStep('UI: Get Edit/Schedule delivery inputs values')
+  async getInputValues(): Promise<IDeliveryOptionsUI> {
+    const [condition, finalDate, location, country, city, street, house, flat] = await Promise.all([
+      this.typeInput.inputValue(),
+      this.dateInputField.inputValue(),
+      (await this.locationInput.isVisible()) && this.locationInput.inputValue(),
+      this.countryInput.inputValue(),
+      this.cityInput.inputValue(),
+      this.streetInput.inputValue(),
+      this.houseInput.inputValue(),
+      this.flatInput.inputValue(),
+    ]);
+    const address = {
+      country: (country as COUNTRIES) || undefined,
+      city: city || undefined,
+      street: street || undefined,
+      house: house ? +house : undefined,
+      flat: flat ? +flat : undefined,
+    };
+    return {
+      condition: (condition as DELIVERY_CONDITIONS) || undefined,
+      finalDate: finalDate || undefined,
+      location: (location as IDeliveryOptionsUI['location']) || undefined,
+      address,
+    };
+  }
+
+  @logStep('UI: Click on Save Delivery button')
+  async clickSave() {
+    await this.saveBtn.click();
+  }
+
+  @logStep('UI: Click on Cancel button')
+  async clickCancel() {
+    await this.cancelBtn.click();
+  }
+
+  @logStep('UI: Get validation errors from Edit/Schedule delivery form')
+  async getFormErrors() {
+    return {
+      country: (await this.countryError.isVisible()) ? await this.countryError.innerText() : null,
+      city: (await this.cityError.isVisible()) ? await this.cityError.innerText() : null,
+      street: (await this.streetError.isVisible()) ? await this.streetError.innerText() : null,
+      house: (await this.houseError.isVisible()) ? await this.houseError.innerText() : null,
+      flat: (await this.flatError.isVisible()) ? await this.flatError.innerText() : null,
+    };
+  }
+
+  @logStep('UI: Check if address inputs are readonly')
+  async areAddressInputsReadonly(includeCountry = true): Promise<boolean> {
```

Comment:
Предложил бы отталкиваться от логики и завязаться на `Location`
```suggestion
  async areAddressInputsReadonly(location: IDeliveryOptionsUI['location'] = 'Home'): Promise<boolean> {
  if (location === 'Home')
```

Reply:
Не очень, потому что локейшен есть не всегда. Например если выбран Delivery Type = Pickup, то локейшена нет совсем. И вот у пикапа как раз редактируемая только страна. А вот если деливери тайп = деливери, то там уже в зависимости от локации. Не хочется сюда такую логику накручивать, выглядит ту мач